src/software/new_geom/util/intersects.cpp

```diff
@@ -0,0 +1,25 @@
+#include "software/new_geom/util/intersects.h"
+
+bool intersects(const Polygon& poly, const Segment& segment)
+{
+    for (const auto& seg : poly.getSegments())
+    {
+        if (::intersects(seg, segment))
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool intersects(const Polygon& poly, const Ray& ray)
+{
+    for (const auto& seg : poly.getSegments())
+    {
+        if (::intersects(seg, ray))
```

Comment:
Where do the implementations for the `intersects` methods we're using here live?

Reply:
It lives in geom/util.h and geom/util.cpp

Reply:
Ah, so will those eventually end up in this file? Or is there a reason for keeping some `intersects` methods in `util`?

Reply:
I believe they will eventually be moved to this file once we take a look at refactoring ```geom/util``` - I'm not sure if @jsn0716 has other plans for it

Reply:
Yeah will probably try to group those util functions like `intersects` into one PR